cmd/sdk-server/main.go

```diff
@@ -128,28 +108,101 @@ func main() {
 
 		var s *gameservers.SDKServer
 		s, err = gameservers.NewSDKServer(viper.GetString(gameServerNameEnv), viper.GetString(podNamespaceEnv),
-			healthDisabled, healthTimeout, healthFailureThreshold, healthInitialDelay, kubeClient, agonesClient)
+			ctlConf.HealthDisabled, ctlConf.HealthTimeout, ctlConf.HealthFailureThreshold,
+			ctlConf.HealthInitialDelay, kubeClient, agonesClient)
 		if err != nil {
 			logger.WithError(err).Fatalf("Could not start sidecar")
 		}
-		ctx, cancel := context.WithCancel(context.Background())
-		defer cancel()
 
 		go s.Run(ctx.Done())
 		sdk.RegisterSDKServer(grpcServer, s)
 	}
 
+	run(ctx, grpcServer, lis, grpcEndpoint, mux, httpServer)
```

Comment:
More a matter of style, but I prefer making functions synchronous and then prefacing them with the `go` keyword.

Reply:
Good call. splitting this out into two synchronous functions reads a lot better.